components/policies/staging/base/konflux-rbac/set-tenant-namespace-visibility/.chainsaw-test/chainsaw-test.yaml

```diff
@@ -0,0 +1,221 @@
+   name: change-namespace-from-private-to-public
+spec:
+  description: |
+    tests that a Namespace with a `konflux-ci.dev/type=tenant` label
+    and no RoleBinding for the `system:authenticated group` will
+    get labeled with `konflux-ci.dev/visibility=private`. Once that
+    Rolebinding is created, the label should change to 
+    `konflux-ci.dev/visibility=public`.
+  bindings:
+  - name: type
+    value: tenant
+  steps:
+  - name: given-kyverno-has-permission-on-resources
+    try:
+    - apply:
+        file: ../kyverno_rbac.yaml
+  - name: apply-set-tenant-namespace-visibility-clusterpolicy
+    try:
+    - apply:
+        file: ../set-tenant-namespace-visibility-clusterpolicy.yaml
+  - name: when-tenant-namespace-exists
+    try:
+    - apply:
+        file: ./resources/tenant-namespace.yaml
+  - name: then-visibility-is-private
+    try:
+    - assert:
+        file: ./resources/expected-private-tenant-namespace.yaml
+  - name: when-authenticated-rolebinding-exists
+    try:
+    - apply:
+        file: ./resources/authorized-rolebinding.yaml
+  - name: then-visibility-is-public
+    try:
+    - assert:
+        file: ./resources/expected-public-tenant-namespace.yaml
+
+---
+
+# yaml-language-server: $schema=https://raw.githubusercontent.com/kyverno/chainsaw/main/.schemas/json/test-chainsaw-v1alpha1.json
+apiVersion: chainsaw.kyverno.io/v1alpha1
+kind: Test
+metadata:
+   name: change-namespace-from-public-to-private
+spec:
+  description: |
+    tests that a Namespace with a `konflux-ci.dev/type=tenant` label
+    and a RoleBinding for the `system:authenticated group` will
+    get labeled with `konflux-ci.dev/visibility=public`. Once that
+    RoleBinding is removed, the label should change to 
+    `konflux-ci.dev/visibility=private`.
+  bindings:
+  - name: type
+    value: tenant
+  steps:
+  - name: given-kyverno-has-permission-on-resources
+    try:
+    - apply:
+        file: ../kyverno_rbac.yaml
+  - name: apply-set-tenant-namespace-visibility-clusterpolicy
+    try:
+    - apply:
+        file: ../set-tenant-namespace-visibility-clusterpolicy.yaml
+  - name: when-tenant-namespace-exists
+    try:
+    - apply:
+        file: ./resources/tenant-namespace.yaml  
+  - name: when-authenticated-rolebinding-exists
+    try:
+    - apply:
+        file: ./resources/authorized-rolebinding.yaml
+  - name: then-visibility-is-public
+    try:
+    - assert:
+        file: ./resources/expected-public-tenant-namespace.yaml
+  - name: when-authenticated-rolebinding-is-deleted
+    try:
+    - delete:
+        file: ./resources/authorized-rolebinding.yaml
+  - name: then-visibility-is-private
+    try:
+    - assert:
+        file: ./resources/expected-private-tenant-namespace.yaml
+
+---
+
+# yaml-language-server: $schema=https://raw.githubusercontent.com/kyverno/chainsaw/main/.schemas/json/test-chainsaw-v1alpha1.json
+apiVersion: chainsaw.kyverno.io/v1alpha1
+kind: Test
+metadata:
+   name: change-namespace-type-to-public-tenant
```

Comment:
this is failing in CI. I tested it locally and it seems to work fine. Probably CI is not powerful enough to complete it on time. I'd suggest increasing timeouts for steps that fails.